Add docs for ALFAcase file (#124)
ESSS/alfasim-sdk: docs/source/alfacase/02_alfacase_syntax.rst

.. _alfacase-syntax:

ALFAcase Syntax
===============

An :program:`ALFAcase` file is a text-based file that must be written following the YAML syntax, only a restricted subset of the YAML specification

> is followed and the following section and this section covers the support YAML syntax how to configure a project.

Reviewer: ```suggestion
is supported and the this section and the next cover the differences.
```

But I'm not sure we should add these here: while I appreciate the intention, I think a link to the Strict YAML docs is more effective and can replace this entire section.

Author: I don't know if a link to the StrictYAML would be good for our users since the StrictYAML docs are focused on the DEV using the lib, with mentions and instructions on how to extract and manipulate the data, many details that out of the scope of our user that only needs to fill a file.

I also don't wanna link a cheatsheet with YAML syntax since only a subset of the specification is followed by strictyaml,  and in my view, our users just need to know how to write a "string", "number", "boolean" and a "list with numbers".

Reviewer: > I don't know if a link to the StrictYAML would be good for our users since the StrictYAML docs are focused on the DEV using the lib

Ahh good point, I misremembered the docs, I thought they had a quick YAML primer but they don't.

> I also don't wanna link a cheatsheet with YAML syntax since only a subset of the specification is followed by strictyaml, and in my view, our users just need to know how to write a "string", "number", "boolean" and a "list with numbers".

I understand, but I also don't want to replicate YAML's documentation here. But OK, I also couldn't find any primer/tutorial which was generic enough, so let's keep it. 👍

The next sections go deep on the syntax, showing different ways to fill some values.

.. _string-alfacase-schema:

String
------

.. code-block:: yaml
   :caption: string syntax

    # key: value
    material_name: Another value goes here.

    # It is possible to put quotes in a string, but it is not necessary
    material_name: 'A string, enclosed in quotes.'

    # variable: variable
    key with spaces: value

.. _number-alfacase-schema:

Number
-------

.. code-block:: yaml
   :caption: number syntax

    # Integer
    value: 100

    # Float
    value: 1.5

    # Scientific Notation
    value: 1e+12

.. _list-alfacase-schema:

List
----

``list`` is a sequence of values and on :program:`ALFAcase` list is denoted by a series of dashes (``-``)
It is possible to define a list in a compressed inserting the value between brackets (``[`` ``]``)

.. code-block:: yaml
   :caption: list syntax

    values:
      - 1
      - 2
      - 3

    # Flow style
    values: [ 1, 2 , 3 ]

.. _bool-alfacase-schema:

Bool
----

``bool`` is case-insensitive and accepts the following options:

    .. code-block:: yaml
        :caption: boolean syntax

>         enable_solver_caching: True     # True

Reviewer 2: I would just group up True and False values together instead of mixing them.

Author: Yep, good suggestion =)
